Mage.Server/src/main/java/mage/server/Main.java

```diff
@@ -358,7 +371,14 @@ public void setMBeanServer(MBeanServer server) {
 
         @Override
         public void setInvoker(ServerInvoker invoker) {
-            ((BisocketServerInvoker) invoker).setSecondaryBindPort(managerFactory.configSettings().getSecondaryBindPort());
+            if (managerFactory.configSettings().isMultiHome()) {
```

Comment:
This guard is what guarantees backward compatibility. If the server is configured as multi-home the new connection string is calculated, otherwise the previous one is configured.